src/Navbar.js

```diff
@@ -107,103 +116,55 @@ const Navbar = React.createClass({
     classes[tbsUtils.prefix(this.props, INVERSE)] = this.props.inverse;
     classes[tbsUtils.prefix(this.props, DEFAULT)] = !this.props.inverse;
 
-    const showHeader =
-      (brand || toggleButton || toggleNavKey != null) &&
-      !this.hasNavBrandChild();
-
     return (
       <ComponentClass {...props} className={classNames(className, classes)}>
         <Grid fluid={fluid}>
-          {showHeader ? this.renderBrandHeader() : null}
-          {ValidComponentChildren.map(children, this.renderChild)}
+          { children }
         </Grid>
       </ComponentClass>
     );
-  },
-
-  renderBrandHeader() {
-    let {brand} = this.props;
-    if (brand) {
-      brand = <NavBrand>{brand}</NavBrand>;
-    }
-
-    return this.renderHeader(brand);
-  },
-
-
-  renderHeader(brand) {
-    const hasToggle = this.props.toggleButton || this.props.toggleNavKey != null;
-    const headerClass = tbsUtils.prefix(this.props, 'header');
+  }
+});
 
-    return (
-      <div className={headerClass}>
-        {brand}
-        {hasToggle ? this.renderToggleButton() : null}
-      </div>
-    );
-  },
+const NAVBAR_STATES = [DEFAULT, INVERSE];
 
-  renderChild(child, index) {
-    const key = child.key != null ? child.key : index;
+Navbar = bsStyles(NAVBAR_STATES, DEFAULT,
+  bsClasses('navbar',
+    uncontrollable(Navbar, { navExpanded: 'onToggle' })
+  )
+);
 
-    if (child.props.bsRole === 'brand') {
-      return React.cloneElement(this.renderHeader(child), {key});
-    }
+function createSimpleWrapper(tag, suffix, displayName) {
+  let wrapper = (
```

Comment:
Not in these cases since none of these components exist yet...I can't think of a reasonable use case for why you'd want to ref these, In anycase I only did it b/c it was terser, not cost in changing it